Classes/Aspects/AugmentationAspect.php

```diff
@@ -193,16 +193,42 @@ public function editableElementAugmentation(JoinPointInterface $joinPoint)
 
     /**
      * @param NodeInterface $node
-     * @param boolean $renderCurrentDocumentMetadata
-     * @return boolean
+     * @param bool $renderCurrentDocumentMetadata
+     * @return bool
      * @throws IllegalObjectTypeException
      */
-    protected function needsMetadata(NodeInterface $node, $renderCurrentDocumentMetadata)
+    protected function needsMetadata(NodeInterface $node, bool $renderCurrentDocumentMetadata): bool
```

Comment:
Currently trying to get more insights here, I think we shouldn't only look at inline editable properties, as we probably need the metadata also if only inspector properties are allowed. And at that point I am really not sure anymore if the performance gains are still there or if we maybe just skip that check altogether and just check for isInBackend() but that is TBD.